### DISCUSSION TITLE
Add Filename Option for Reblog Name

### DISCUSSION THREAD

**Discussion by IntrovertedAB** [TumblThreeApp/TumblThree#496] [Q&A (Question & Answer)]
I've noticed that when there is a reblog download that the images.txt file includes the name of the account that it was reblogged from.  Is there a way to add that account name to the file renaming system?  It looks like the app already captures the information so I'd imagine it wouldn't be too hard to implement

**Comment by IntrovertedAB** [comment]
After doing some searching around the repository, it looks like this may have been implemented for the Twitter Crawler as %o.  I'm not a coder though, so I'm not really sure. If so, then I'd imagine the change could be even easier.

**Comment by IntrovertedAB** [comment]
New update -- %o actually works a fair amount of the time, but not always.

**Comment by IntrovertedAB** [comment]
Okay, here's the desired functionality, let me know if it is possible:

For any file associated with a reblog, I'd like to be able to modify the file name so that the original poster's name (even if deactivated) is included into it.  So, for example, if one were to try to download the image from this post (https://www.tumblr.com/purpleclodhumanmonger/665182581175828480), the f-o-t-o-b-l-o-g username would return because it was the original poster, rather than the direct reblog, which was from erikalnieto.

**Reply by thomas694** [reply]
I added the blog with the filename template `%o _ %f`, downloaded that day and got this file:
f-o-t-o-b-l-o-g _ tumblr_lp60wiHnNy1qh7487o1_r3_500.gif
What filename did you get?

Technically the option `%o` uses the origin/root blog name of the reblogged post, and if it's empty, the blog name it was currently reblogged from.

> New update -- %o actually works a fair amount of the time, but not always.

If something doesn't work, we need an example and the steps how you got there.

**Reply by IntrovertedAB** [reply]
Hi there, thanks for responding.  I did some additional testing, and I found the issue. For blogs that do Porn, TumblThree is downloading two versions of the image. One is just a string of characters, e.g. f370231b7ed46633ba3a766f189385f56f36d9b9.jpg, which is typically lower resolution or the same resolution as the second one, which includes the original blog's name. 

So, for example, if you were to download from kung-wu.tumblr.com, which is marked as mature/hidden but only reblogs pictures of girls in sweaters, for the 30 or so seconds that I let it run without a date filter (only covered posts in November for me) then there was only one version of each picture downloaded and all had the original blog's name there.  If you go to [snip], which is mostly pictures of women's breasts, and let it run without a date filter for ~30 seconds (which for me got sometime into October), then you get the duplicate file issue I described above.

Part of what had thrown me off before was that I was using AllDup to try to check for duplicates and it looks like I was too strict in the matching criteria (I think I put 97%) but since there can be substantial resolution differences between the two files I was missing out on some of them.

I don't know whether this is an actual issue, or if my settings are creating an expected result, so I've attached screenshots of my settings. 

I more thought it interesting that non-hidden/mature but porn had a different outcome than hidden/mature but not really porn.

**Reply by IntrovertedAB** [reply]
It would help if I attached the settings.
![Screenshot 2023-11-19 121122](https://github.com/TumblThreeApp/TumblThree/assets/141250467/9d2f6388-3beb-46f6-8315-31cd47b17186)
![Screenshot 2023-11-19 121208](https://github.com/TumblThreeApp/TumblThree/assets/71355143/9ddab887-03f0-43fd-97a7-81cfd85705c2)
![Screenshot 2023-11-19 121259](https://github.com/TumblThreeApp/TumblThree/assets/141250467/57ed97f7-783e-4e1e-8ad6-f9e35affc113)
![Screenshot 2023-11-19 121321](https://github.com/TumblThreeApp/TumblThree/assets/141250467/6424565e-b8c2-47c7-a1d8-a07d9c7b0ae8)
![Screenshot 2023-11-19 121343](https://github.com/TumblThreeApp/TumblThree/assets/141250467/26b57a5c-257d-4c87-ad13-396f4bd724c6)
